plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/controller/CodeTransformChatController.kt

```diff
@@ -213,28 +286,90 @@ class CodeTransformChatController(
         val moduleVirtualFile: VirtualFile = modulePath.toVirtualFile() as VirtualFile
         val moduleName = context.project.getModuleOrProjectNameForFile(moduleVirtualFile)
 
-        codeTransformChatHelper.addNewMessage(buildUserSelectionSummaryChatContent(moduleName))
+        codeTransformChatHelper.addNewMessage(buildUserLanguageUpgradeSelectionSummaryChatContent(moduleName))
 
         val sourceJdk = getSourceJdk(moduleVirtualFile)
 
         val selection = CustomerSelection(
-            moduleVirtualFile,
-            sourceJdk,
-            JavaSdkVersion.JDK_17,
+            configurationFile = moduleVirtualFile,
+            sourceJavaVersion = sourceJdk,
+            targetJavaVersion = JavaSdkVersion.JDK_17,
         )
 
         // Create and set a session
         codeModernizerManager.createCodeModernizerSession(selection, context.project)
 
         // Publish metric to capture user selection before local build starts
-        telemetry.submitSelection("Confirm", selection)
+        telemetry.submitSelection("Confirm-Java", selection)
 
         codeTransformChatHelper.run {
             addNewMessage(buildUserInputSkipTestsFlagChatIntroContent())
             addNewMessage(buildUserInputSkipTestsFlagChatContent())
         }
     }
 
+    override suspend fun processCodeTransformSelectSQLModuleSchemaAction(message: IncomingCodeTransformMessage.CodeTransformSelectSQLModuleSchema) {
+        val moduleName = context.project.getModuleOrProjectNameForFile(message.modulePath.toVirtualFile())
+        codeTransformChatHelper.addNewMessage(buildUserSQLConversionSelectionSummaryChatContent(moduleName, message.schema))
+        codeModernizerManager.codeTransformationSession?.let {
+            it.sessionContext.configurationFile = message.modulePath.toVirtualFile()
+            it.sessionContext.schema = message.schema
+        }
+        // start the SQL conversion
+        runInEdt {
+            codeModernizerManager.runModernize()
+        }
+    }
+
+    override suspend fun processCodeTransformSelectSQLMetadataAction(message: IncomingCodeTransformMessage.CodeTransformSelectSQLMetadata) {
+        runInEdt {
+            val descriptor = FileChooserDescriptorFactory.createSingleFileDescriptor()
+                .withDescription("Select metadata file")
+                .withFileFilter { it.extension == "zip" }
+
+            val selectedZipFile = FileChooser.chooseFile(descriptor, null, null) ?: return@runInEdt
+            val extractedZip = createTempDirectory("codeTransformSQLMetadata", null)
+
+            unzipFile(selectedZipFile.toNioPath(), extractedZip.toPath(), true)
+
+            val sctFile = extractedZip.listFiles { file -> file.name.endsWith(".sct") }.firstOrNull()
+
+            val metadataValidationResult = validateSctMetadata(sctFile)
+
+            if (!metadataValidationResult.valid) {
+                CoroutineScope(EDT).launch {
```

Comment:
also this breaks structured concurrency, should be `withContext(...)`

Reply:
Removed entirely in favor of `withContext`